### ISSUE TITLE
[staging & production] SHARE graph has fractions of events

### ISSUE THREAD

**Issue by caileyfitz** [CenterForOpenScience/osf.io#2326]
# Steps
1. Go to the SHARE page on staging or production
2. Enter "phosphatase activity in matrix vesicles as determined by infrared spectroscopy" into the search bar
3. Press search
4. Confirm there is only one result
4. Look at the graph of results

# Expected
There is only one result, so I would expect that the graph of results would show that result with correct increments. Since the y-axis is labeled as "Number of Events," I expect the scale to include only positive whole numbers.

# Actual
The graph includes only one result, but the y-axis is labeled in increments of tenths of events, which is weird because there can only be whole numbers of events. Additionally, sometimes the graph is drawn to show a negative number of events or a fraction of events, even though the data points represent only whole numbers.

Example with graph appearing to show negative results (staging):
![screen shot 2015-03-26 at 12 22 44 pm](https://cloud.githubusercontent.com/assets/5975444/6852395/c9a68824-d3ba-11e4-97e4-8e5c0e2d0466.png)

Example with graph appearing to show a fraction of number of events (production):
![screen shot 2015-03-26 at 1 13 25 pm](https://cloud.githubusercontent.com/assets/5975444/6852428/ecdd3220-d3ba-11e4-9552-06b1ee11e386.png)

**Comment by nanchen47** [comment]
@caileyfitz I can't repo this problem on staging or production. Can you verify if this problem still exists? Thanks.

**Comment by caileyfitz** [comment]
@nanchen47 Yes, this still occurs. You have to search with the quotation marks: *"phosphatase activity in matrix vesicles as determined by infrared spectroscopy"*

**Comment by brianjgeiger** [comment]
https://osf.io/share/?q=%22phosphatase+activity+in+matrix+vesicles+as+determined+by+infrared+spectroscopy%22&sort=Relevance

**Comment by lbanner** [comment]
Could be wrong, but I am doubting that graph dynamically generates the axis labels -- and so this would be difficult to overcome -- I'd suggest we accept this edge case for now and close -- Debate?

**Comment by brianjgeiger** [comment]
Probably not too difficult. http://stackoverflow.com/questions/12643591/how-to-limit-d3-svg-axis-to-integer-labels

**Comment by lbanner** [comment]
OK - Thank you @brianjgeiger -- will ping @erinspace and @fabianvf to put on their radar, and decide priority on this one.

**Comment by erinspace** [comment]
On the radar! Thanks all!

**Comment by samanehsan** [comment]
Closing this since the graph now displays correctly: 
![screen shot 2015-06-18 at 10 47 02 am](https://cloud.githubusercontent.com/assets/6414394/8234034/a26fcad4-15a7-11e5-93cf-140299b75444.png)

**Comment by fabianvf** [comment]
The behavior is actually the same as when the bug was reported, the issues was with the y-axis ticks displaying fractions of events, when fractions of events cannot exist.

**Comment by samanehsan** [comment]
Oh okay, in that case I will reopen. I thought the issue was that the graph wasn't displaying correctly -- before, the graph showed negative results and appeared to be a fraction of an event but now it correctly displays as one event.

**Comment by caileyfitz** [comment]
I am also still seeing that the graph shows negative events:
![share graph](https://cloud.githubusercontent.com/assets/5975444/9407675/7c5de97a-47d9-11e5-821a-f26fb7742938.png)
